Sources/Sentry/SentrySessionTaskSearch.m

```diff
@@ -0,0 +1,57 @@
+/**
+ * Part of this code was copied from
+ * https://github.com/AFNetworking/AFNetworking/blob/4eaec5b586ddd897ebeda896e332a62a9fdab818/AFNetworking/AFURLSessionManager.m#L407-L418
+ * under the MIT license
+ */
+
+#import "SentrySessionTaskSearch.h"
+#import <objc/runtime.h>
+
+@implementation SentrySessionTaskSearch
+
++ (NSArray<Class> *)urkSessionTaskClassesToTrack
+{
+
+    /**
+     * In order to be able to track a network request, we need to know when it starts and when it
+     * finishes. NSURLSessionTask has a `resume` method that starts the request, and the only way to
+     * know when it finishes is to check the task `state`. Using KVO is not working, so we are
+     * swizzling `setState:`. Depending on the iOS version NSURLSessionTask does not implements
+     * `setState:` and Apple uses a subclass returned by NSURLSession that implementes `setState:`.
+     * We need to discover which class to swizzle.
+     *
+     * Apple intermediate class for iOS does not call [super resume], so we can swizzle both class.
+     * This Apple approach may change in the future, we need to have enough tests to detect it
+     * early.
+     */
+
+    NSURLSessionConfiguration *configuration =
+        [NSURLSessionConfiguration ephemeralSessionConfiguration];
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration];
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnonnull"
+    NSURLSessionDataTask *localDataTask = [session dataTaskWithURL:nil];
+#pragma clang diagnostic pop
+    Class currentClass = [localDataTask class];
+    NSMutableArray *result = [[NSMutableArray alloc] init];
```

Comment:
`l`: I would like to have a linter warning us about not using generics with arrays, so I don't have to point this out 😄 
```suggestion
    NSMutableArray<Class> *result = [NSMutableArray new];
```

Reply:
A linter would be nice. 
Old habits. I believe objc didn't have generics when I start working with it.
I'll police myself.